extensions/git/src/protocolHandler.ts

```diff
@@ -38,7 +38,18 @@ export class GitProtocolHandler implements UriHandler {
 
 		let cloneUri: Uri;
 		try {
-			cloneUri = Uri.parse(Array.isArray(data.url) ? data.url[0] : data.url, true);
+			let rawUri = Array.isArray(data.url) ? data.url[0] : data.url;
+
+			// Handle SSH Uri
+			// Ex: git@github.com:microsoft/vscode.git
+			if (/^git@[^\/:]+:/i.test(rawUri)) {
+				rawUri = rawUri.replace(/:/, '/');
+				rawUri = `ssh://${rawUri}`;
+			}
```

Comment:
How about a simplification?

```suggestion
			rawUri = rawUri.replace(/^(git@[^\/:]+)(:)/i, 'ssh://$1/');
```